pkg/istio-agent/agent.go

```diff
@@ -260,7 +260,10 @@ func (a *Agent) initializeEnvoyAgent() error {
 	a.envoyWaitCh = make(chan error, 1)
 	if a.cfg.EnableDynamicBootstrap {
 		// Simulate an xDS request for a bootstrap
+		a.wg.Add(1)
```

Comment:
Agent.Run() calls wg.Done() which blocks until all spawned goroutines exit. Speaking of which, the race indicates that xDS proxy escapes its inputs into goroutines that don't exit when handle finishes. This is sketchy since we can accidentally write to inputs at random times. I fixed this by creating a fresh input every time.

Reply:
wg.Done doesn't block, wg.Wait does, which I don't see called anywhere?

Reply:
Sorry, I meant wg.Wait. See down below, line 418.

Reply:
ah, the git diff confused me. thanks